### ISSUE TITLE
HeaderSecurity overhead

### ISSUE THREAD

**Issue by mindplay-dk** [zendframework/zend-diactoros#45]
Ran across this while doing some performance analysis with xdebug and cachegrind.

Looks like HeaderSecurity overhead is pretty substantial:

![image](https://cloud.githubusercontent.com/assets/103348/7961954/e43eff14-0a0a-11e5-898d-7eb9d35570a1.png)

Do we need this (relatively expensive) header security when constructing a request model from superglobals? Doesn't the server take care of vetting incoming requests before it dispatches PHP?

**Comment by weierophinney** [comment]
Yes, it needs to be done. Please read this: http://framework.zend.com/security/advisory/ZF2015-04

While the server _should_ be vetting it properly, we cannot be certain what server you're behind, and many servers pass through the headers as-is. (As an example, I'm not sure how React handles this.)

If you have ideas on how to *safely* bypass the checks, please provide a proof-of-concept.

**Comment by mtymek** [comment]
If this is necessary, we can at least ease the pain using regular expressions ;-)

I just made a quick experiment, replaced loop from `HeaderSecurity::isValid` with following regexp:

```php
    if (preg_match('/[^\x09\x0a\x0d\x20-\x7E\x80-\xFE]/', $value)) {
        return false;
    }
```

Before:

![image](https://cloud.githubusercontent.com/assets/777893/7963258/0f08caa2-0a12-11e5-970d-4d63251c8353.png)

After:

![image](https://cloud.githubusercontent.com/assets/777893/7963267/1b9498b4-0a12-11e5-8346-c236fa33c704.png)


It was just quick experiment, but result is promising. I will prepare PR somewhere tomorrow.

**Comment by Maks3w** [comment]
https://github.com/zendframework/zend-diactoros/pull/47 fix this?

**Comment by weierophinney** [comment]
Closed with #47

**Comment by mindplay-dk** [comment]
Nice work, guys :-)
